### PR TITLE
Added cli tests for module stream 6.5 feature

### PR DESCRIPTION
**Pull request by omkarkhatavkar**
**These are the new cli tests for module streams list and info.** 

**Test Result** 

```
(env) [root@okhatavk robottelo]# pytest tests/foreman/cli/test_repository.py -v -k "module_stream"
============================================================================ test session starts ============================================================================
platform linux -- Python 3.4.8, pytest-3.6.1, py-1.7.0, pluggy-0.6.0 -- /home/okhatavk/Satellite/robottelo/env/bin/python3.4
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/okhatavk/Satellite/robottelo, inifile:
plugins: xdist-1.23.2, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.6.0
collecting 89 items                                                                                                                                                         2018-10-30 15:07:45 - conftest - DEBUG - BZ deselect is disabled in settings

collected 89 items / 86 deselected                                                                                                                                          

tests/foreman/cli/test_repository.py::RepositoryTestCase::test_module_stream_info_validation PASSED                                                                   [ 33%]
tests/foreman/cli/test_repository.py::RepositoryTestCase::test_module_stream_list_validation PASSED                                                                   [ 66%]
tests/foreman/cli/test_repository.py::RepositoryTestCase::test_positive_create_get_update_delete_module_streams PASSED                                                [100%]

================================================================= 3 passed, 86 deselected in 262.00 seconds =================================================================
(env) [root@okhatavk robottelo]# pytest tests/foreman/cli/test_contentview.py -v -k "module_stream"
============================================================================ test session starts ============================================================================
platform linux -- Python 3.4.8, pytest-3.6.1, py-1.7.0, pluggy-0.6.0 -- /home/okhatavk/Satellite/robottelo/env/bin/python3.4
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/okhatavk/Satellite/robottelo, inifile:
plugins: xdist-1.23.2, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.6.0
collecting 101 items                                                                                                                                                        2018-10-30 15:50:27 - conftest - DEBUG - BZ deselect is disabled in settings

collected 101 items / 100 deselected                                                                                                                                        

tests/foreman/cli/test_contentview.py::ContentViewTestCase::test_positive_publish_custom_content_module_stream PASSED                                                 [100%]

================================================================ 1 passed, 100 deselected in 158.27 seconds =================================================================
(env) [root@okhatavk robottelo]# 
```

### CODE REVIEW

File: tests/foreman/cli/test_contentview.py [hunk 1]
```diff
@@ -2201,6 +2204,59 @@ def test_positive_publish_custom_content(self):
...
+            'Module Streams are not associated with Content View')
+        # Publish another new version of CV
+        ContentView.add_repository({
+            u'id': new_cv['id'],
+            u'repository-id': software_repo['id'],
+        })
+        ContentView.publish({u'id': new_cv['id']})
+        new_cv_version_2 = ContentView.info({u'id': new_cv['id']})['versions'][1]
+        module_streams = ModuleStream.list({'content-view-version-id': new_cv_version_2['id']})
+        self.assertEqual(
```
Comment (ntkathole) [inline tests/foreman/cli/test_contentview.py]: @omkarkhatavkar Will there always be 12 modules streams at given url? What about  using `self.assertGreater()`

Comment (omkarkhatavkar) [reply tests/foreman/cli/test_contentview.py]: @ntkathole make sense! thanks

File: tests/foreman/cli/test_repository.py [hunk 2]
```diff
@@ -1790,6 +1793,158 @@ def test_positive_upload_content_srpm(self):
...
+        Repository.synchronize({'id': repo2['id']})
+        actual_result = ModuleStream.info({u'id': "12"})
+        expected_result = {
+            'module-stream-name': 'nodejs',
+            'stream': '8',
+            'architecture': 'noarch',
+            'artifacts': [
+                {
+                    'id': '27',
+                    'name': 'nodejs-1:8.10.0-3.module_1572+d7ec111e.x86_64'
```
Comment (ntkathole) [inline tests/foreman/cli/test_repository.py]: If we use such values, we might need to update testcase everytime. Can we have some better way?

File: tests/foreman/cli/test_contentview.py (hunk 1 above)
Comment (pondrejk) [reply tests/foreman/cli/test_contentview.py]: yes, will these repos stay unchanged for us in the future? If not sure, one possibility is to create your own copy on fedorapeople.org

File: tests/foreman/cli/test_contentview.py [hunk 3]
```diff
@@ -2201,6 +2204,59 @@ def test_positive_publish_custom_content(self):
             'Publishing new version of CV was not successful',
         )
 
+    @tier2
+    @run_only_on('sat')
```
Comment (pondrejk) [inline tests/foreman/cli/test_contentview.py]: I think  @run_only_on('sat') is deprecated, no harm if you remove it

Comment (omkarkhatavkar) [reply tests/foreman/cli/test_contentview.py]: @pondrejk thanks will remove that :)

File: tests/foreman/cli/test_repository.py (hunk 2 above)
Comment (omkarkhatavkar) [reply tests/foreman/cli/test_repository.py]: @ntkathole not everytime unless until there change in modulemd file or change in the repo, which supposed to be infrastructure change can be controlled. let me know your thoughts?

File: tests/foreman/cli/test_repository.py [hunk 4]
```diff
@@ -1790,6 +1793,158 @@ def test_positive_upload_content_srpm(self):
...
+        # adding repo with same yum url should not change count.
+        duplicate_repo = self._make_repository({
+            u'content-type': u'yum',
+            u'url': CUSTOM_MODULE_STREAM_REPO_1,
+        })
+        Repository.synchronize({'id': duplicate_repo['id']})
+
+        module_streams = ModuleStream.list({'repository-id': repo['id']})
+        self.assertEqual(len(module_streams), 5,
+                         'Module Streams get worked Correctly')
```
Comment (pondrejk) [inline tests/foreman/cli/test_repository.py]: nitpick "correctly" :)

File: tests/foreman/cli/test_repository.py (hunk 2 above)
Comment (omkarkhatavkar) [reply tests/foreman/cli/test_repository.py]: @ntkathole @pondrejk I understood that i need to make sure that repos should be hosted somewhere in QA infra.

File: robottelo/constants.py [hunk 5]
```diff
@@ -529,6 +529,12 @@
 CUSTOM_RPM_REPO = (
     u'http://repos.fedorapeople.org/repos/pulp/pulp/fixtures/rpm/'
 )
+CUSTOM_MODULE_STREAM_REPO_1 = (
+    u'http://mirror.math.princeton.edu/pub/fedora/linux/releases/28/Modular/x86_64/os/'
```
Comment (ldjebran) [inline robottelo/constants.py]: This is an external url, are we sure it will be always available

Comment (parthaa) [reply robottelo/constants.py]: @omkarkhatavkar I am guessing
```https://dl.fedoraproject.org/pub/fedora/linux/updates/29/Modular/x86_64/```
or 
```https://repos.fedorapeople.org/pulp/pulp/fixtures/rpm-with-modules/``` is better

Comment (omkarkhatavkar) [reply robottelo/constants.py]: @parthaa thanks for confirming, I will update the PR with the same changes.

Comment (omkarkhatavkar) [reply robottelo/constants.py]: @ldjebran @parthaa and @pondrejk  updated all code based on new repos. please have a review

File: tests/foreman/cli/test_repository.py [hunk 6]
```diff
@@ -1790,6 +1793,147 @@ def test_positive_upload_content_srpm(self):
...
+        })
+        Repository.synchronize({'id': repo['id']})
+        repo = Repository.info({'id': repo['id']})
+        self.assertEqual(repo['content-counts']['module-streams'], '7',
+                         'Module Streams not synced correctly')
+
+        # adding repo with same yum url should not change count.
+        duplicate_repo = self._make_repository({
+            u'content-type': u'yum',
+            u'url': CUSTOM_MODULE_STREAM_REPO_1,
```
Comment (ldjebran) [inline tests/foreman/cli/test_repository.py]: perhaps 
```python
'url': CUSTOM_MODULE_STREAM_REPO_2,
```

File: tests/foreman/cli/test_repository.py [hunk 7]
```diff
@@ -1790,6 +1793,147 @@ def test_positive_upload_content_srpm(self):
...
+        product2 = make_product_wait(
+            {u'organization-id': self.org['id']},
+        )
+        repo2 = self._make_repository({
+            u'content-type': u'yum',
+            u'url': CUSTOM_MODULE_STREAM_REPO_2,
+            u'product-id': product2['id']
+        })
+        Repository.synchronize({'id': repo2['id']})
+        actual_result = ModuleStream.info({u'id': "5"})
```
Comment (ldjebran) [inline tests/foreman/cli/test_repository.py]: from where we got this id=5, perhaps we should list first with product-id to get the module steam id and verify here and later later with  that id

File: tests/foreman/cli/test_repository.py [hunk 8]
```diff
@@ -1790,6 +1793,147 @@ def test_positive_upload_content_srpm(self):
...
+        print(actual_result)
+        expected_result = {
+            'module-stream-name': 'walrus',
+            'stream': '5.21',
+            'architecture': 'x86_64',
+            'artifacts': [
+                {
+                    'id': '5',
```
Comment (ldjebran) [inline tests/foreman/cli/test_repository.py]: same as previous comment about the module steam id

File: tests/foreman/cli/test_repository.py (hunk 7 above)
Comment (omkarkhatavkar) [reply tests/foreman/cli/test_repository.py]: @ldjebran I have updated the codebase with change.

File: tests/foreman/cli/test_repository.py (hunk 6 above)
Comment (omkarkhatavkar) [reply tests/foreman/cli/test_repository.py]: @ldjebran update this as well.

File: tests/foreman/cli/test_repository.py [hunk 9]
```diff
@@ -1790,6 +1793,150 @@ def test_positive_upload_content_srpm(self):
...
+        Repository.synchronize({'id': repo2['id']})
+        module_streams = ModuleStream.list({'repository-id': repo2['id']})
+        actual_result = None
+        for module in module_streams:
+            if module['module-stream-name'] == 'walrus' and module['stream'] == '5.21':
+                actual_result = ModuleStream.info({u'id': module['id']})
+                break
+        if actual_result is None:
+            raise Exception('Failed to find the expected module stream in {}'.
+                            format(module_streams))
```
Comment (ldjebran) [inline tests/foreman/cli/test_repository.py]: Recommend :
```python
        module_streams = ModuleStream.list({
            'repository-id': repo2['id'],
            'search': 'name="walrus" and stream="5.21"'
        })
        actual_result = ModuleStream.info({u'id': module_streams[0]['id']})
```

File: tests/foreman/cli/test_repository.py [hunk 10]
```diff
@@ -1790,6 +1793,150 @@ def test_positive_upload_content_srpm(self):
...
+        expected_result = {
+            'module-stream-name': 'walrus',
+            'stream': '5.21',
+            'architecture': 'x86_64',
+         }
+        for (key, value) in expected_result.items():
+            self.assertEqual(actual_result["{}".format(key)], value)
```
Comment (ldjebran) [inline tests/foreman/cli/test_repository.py]: Recommend:
```python
self.assertEqual(
            expected_result,
            {key: value for key, value in actual_result.items() if key in expected_result}
        )
```

Comment (omkarkhatavkar) [reply tests/foreman/cli/test_repository.py]: classic pythonic code :)

**Comment by omkarkhatavkar** [comment]
@ldjebran  Done !

```
root@okhatavk robottelo (modularity_tests_satellite_6.5_feature) $ pytest tests/foreman/cli/test_repository.py -v -k "module_stream"
============================================================================ test session starts ============================================================================
platform linux -- Python 3.4.8, pytest-4.0.1, py-1.7.0, pluggy-0.8.0 -- /home/okhatavk/Satellite/robottelo/env/bin/python3.4
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/okhatavk/Satellite/robottelo, inifile:
plugins: xdist-1.23.2, services-1.3.0, mock-1.10.0, forked-0.2, cov-2.6.0
collecting 0 items                                                                                                                                                          2018-12-05 14:49:17 - conftest - DEBUG - BZ deselect is disabled in settings

collected 89 items / 86 deselected                                                                                                                                          

tests/foreman/cli/test_repository.py::RepositoryTestCase::test_module_stream_info_validation PASSED                                                                   [ 33%]
tests/foreman/cli/test_repository.py::RepositoryTestCase::test_module_stream_list_validation PASSED                                                                   [ 66%]
tests/foreman/cli/test_repository.py::RepositoryTestCase::test_positive_create_get_update_delete_module_streams PASSED                                                [100%]

================================================================= 3 passed, 86 deselected in 475.82 seconds =================================================================

```

**Pull request merged**